src/MeetingsSDKAdapter.js

```diff
@@ -942,31 +939,39 @@ export default class MeetingsSDKAdapter extends MeetingsAdapter {
    *
    * @param {string} ID Meeting ID
    * @param {string} cameraID ID of the camera to switch to
-   * @private
    */
   async switchCamera(ID, cameraID) {
-    const sdkMeeting = this.fetchMeeting(ID);
-    const {stream, permission} = await this.getStream(
-      ID,
-      {sendVideo: true},
-      {video: {deviceId: cameraID}},
-    ).toPromise();
+    await this.updateMeeting(ID, async (meeting) => {
+      let updates;
+      const {stream, permission} = await this.getStream(
+        ID,
+        {sendVideo: true},
+        {video: {deviceId: cameraID}},
+      ).toPromise();
+
+      if (stream) {
+        updates = {
+          localVideo: {
+            stream,
+          },
+          cameraID,
+        };
 
-    if (stream) {
-      this.meetings[ID].localVideo.stream = stream;
-      this.meetings[ID].cameraID = cameraID;
+        if (meeting.state === MeetingState.JOINED) {
+          const sdkMeeting = this.fetchMeeting(ID);
```

Comment:
the updater function can now receive sdkMeeting (in addition to adapter meeting)

Reply:
```js
if (!stream) {
  throw ...;
}

if (joined) {
  updateVideo();
}

return {
  localVideo: {stream},
  cameraID,
};